### PR TITLE
[SYSSETUP] Add new Env section to unattend.inf

### PR DESCRIPTION
**Pull request by learn-more**
This allows to add environment variables during unattended setup

### CODE REVIEW

File: dll/win32/syssetup/wizard.c [hunk 1]
```diff
@@ -2831,6 +2831,53 @@ ProcessUnattendSection(
     }
 
     RegCloseKey(hKey);
+
+    if (SetupFindFirstLineW(pSetupData->hSetupInf,
+        L"Env",
+        NULL,
+        &InfContext))
+    {
+        if (RegCreateKeyExW(
```
Comment (HBelusca) [inline dll/win32/syssetup/wizard.c:2840]: That registry key should already exist; if it doesn't then it should be an error.

File: dll/win32/syssetup/wizard.c [hunk 2]
```diff
@@ -2831,6 +2831,53 @@ ProcessUnattendSection(
     }
 
     RegCloseKey(hKey);
+
+    if (SetupFindFirstLineW(pSetupData->hSetupInf,
```
Comment (HBelusca) [inline dll/win32/syssetup/wizard.c:2835]: Indent seems inconsistent wrt. the rest of the code in this file.

**Review by HBelusca — COMMENTED** [review]
Interesting feature; I wonder how one does the same with Windows.

File: dll/win32/syssetup/wizard.c (hunk 2 above)
Comment (learn-more) [reply dll/win32/syssetup/wizard.c:2835]: Indeed, but I don't feel like manually spacing that stuff out

File: dll/win32/syssetup/wizard.c (hunk 1 above)
Comment (learn-more) [reply dll/win32/syssetup/wizard.c:2840]: The only reason why I would see this fail is when the key cannot be opened for some reason?

**Comment by learn-more** [comment]
@HBelusca do you want to investigate if there is a more windows-like way to do this, or shall I go ahead and commit this?

**Comment by HBelusca** [comment]
> @HBelusca do you want to investigate if there is a more windows-like way to do this, or shall I go ahead and commit this?

I did search around a bit to find whether there was a windows-like way to do this, but I haven't found succeeded to find anything.
So I would say, go ahead and commit the PR, and in the future, if I or someone else is able to find whether Windows has a better way of doing this, one could always fix the code.

(Personally I don't think they have one: unattend.inf is just a *response* file, i.e. to automate giving the answers for the installer, and not a file that contains new settings; otherwise one would have to edit the registry template hive files, for example this seems to be what happens for e.g. WinPE or Windows installation ISOs when they are customized).

**Pull request merged**